pkg/controller/deployment/deployment_controller_test.go

```diff
@@ -0,0 +1,133 @@
+package deployment
+
+import (
+	"sort"
+	"testing"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	"github.com/jaegertracing/jaeger-operator/pkg/inject"
+	"github.com/stretchr/testify/assert"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestSyncOnJaegerChanges(t *testing.T) {
+	// prepare
+	jaeger := v1.NewJaeger(types.NamespacedName{
+		Namespace: "observability",
+		Name:      "my-instance",
+	})
+
+	objs := []runtime.Object{
+		&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{
+			Name: "ns-with-annotation",
+			Annotations: map[string]string{
+				inject.Annotation: "true",
+			},
+		}},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "dep-without-annotation",
+				Namespace: "ns-with-annotation",
+			},
+		},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "dep-with-annotation",
+				Namespace: "ns-with-annotation",
+				Annotations: map[string]string{
+					inject.Annotation: "true",
+				},
+			},
+		},
+
+		&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{
+			Name: "ns-without-annotation",
+		}},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "dep-without-annotation",
+				Namespace: "ns-without-annotation",
+			},
+		},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "dep-with-annotation",
+				Namespace: "ns-without-annotation",
+				Annotations: map[string]string{
+					inject.Annotation: "true",
+				},
+			},
+		},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "dep-with-another-jaegers-label",
+				Namespace: "ns-without-annotation",
+				Labels: map[string]string{
```

Comment:
Should the inject annotation be included here, as in the deployment below - but as it is using a different Jaeger in the label it wouldn't be returned?

Reply:
Well, may be yes to be consistent with the expected states of a deployment, as  I cannot imagine an scenario where we have the label and not the annotation.

On the other side, for what I this is testing it doesn't matter.

https://github.com/jaegertracing/jaeger-operator/pull/1334/files#diff-c553184997fa4c9f63e697af2f39eb2e6fb43eeff64a0ef66c9e955f61347b02R205

Whether this has or not an annotation, the first thing the `syncOnJaegerChanges` checks, is for the label, so it doesn't matter if we don't have an annotation.

Reply:
Agree, it was more to reflect a realistic deployment to avoid any failures later if the logic was to change.

Reply:
In that case. Yes , it should be added :)

Reply:
> Should the inject annotation be included here, as in the deployment below - but as it is using a different Jaeger in the label it wouldn't be returned?

I left only the `Label` here as that's the only thing the code for this case will use. If we think we need to check both the annotation and the label, then both the test and the code need to be changed.

Reply:
I don't think we need to check both, Is more that the case should reflect a real deployment. 

I understand that this test is not using the annotation, for this particular case it only check for the label. But if in the future we change this logic it could help to have the deployment as close as real as possible.